test/e2e/break_glass_credentials_test.go

```diff
@@ -0,0 +1,114 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestBreakGlassCredentials(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithCancel(testContext)
+	defer cancel()
+
+	clusterOpts := globalOpts.DefaultClusterOptions(t)
+
+	e2eutil.NewHypershiftTest(t, ctx, func(t *testing.T, g Gomega, mgtClient crclient.Client, hostedCluster *hyperv1.HostedCluster) {
+		// Sanity check the cluster by waiting for the nodes to report ready
+		t.Logf("Waiting for guest client to become available")
+		_ = e2eutil.WaitForGuestClient(t, ctx, mgtClient, hostedCluster)
+
+		// Grab the break-glass client certificate signer
+		clientSigner := manifests.CustomerSystemAdminSignerCA(hostedCluster.Namespace)
+		if err := wait.PollUntilContextTimeout(ctx, 1*time.Second, 30*time.Minute, true, func(ctx context.Context) (done bool, err error) {
+			getErr := mgtClient.Get(ctx, crclient.ObjectKeyFromObject(clientSigner), clientSigner)
+			if errors.IsNotFound(getErr) {
+				return false, nil
+			}
+			return getErr == nil, err
+		}); err != nil {
+			t.Errorf("client signer didn't become available: %v", err)
+		}
+
+		// Grab the break-glass client certificate
+		clientCertificate := manifests.CustomerSystemAdminClientCertSecret(hostedCluster.Namespace)
```

Comment:
clientCertificate is fetched here but seems unused, shouldn't this be used in the kubeconfig below instead of the signer?